src/variant_object.cpp

```diff
@@ -365,15 +365,35 @@ namespace fc
       return *this;
    }
 
-   limited_mutable_variant_object::limited_mutable_variant_object( uint32_t m )
-         : mutable_variant_object(), _max_depth(m - 1)
+   limited_mutable_variant_object::limited_mutable_variant_object( uint32_t m, bool skip_on_exception )
+         : mutable_variant_object(),
+           _max_depth(m - 1),
+           _reached_depth_limit(m == 0),
+           _skip_on_exception(skip_on_exception)
    {
-      FC_ASSERT( m > 0, "Recursion depth exceeded!" );
+      if( !skip_on_exception )
+         FC_ASSERT( m > 0, "Recursion depth exceeded!" );
+      else if( m == 0 )
+         set( "__err_msg", "[ERROR: Recusion depth exceeded!]" );
    }
 
    limited_mutable_variant_object& limited_mutable_variant_object::operator()( const variant_object& vo )
    {
-      mutable_variant_object::operator()( vo );
+      if( _reached_depth_limit )
```

Comment:
I don't think modifying this is necessary, but doesn't hurt either.
(vo is basically a map string->variant, which is 1 level deep - no recursion into sub-variant etc.)